backend/app/Http/Controllers/TransactionController.php

```diff
@@ -25,4 +26,88 @@ public function show(Transaction $transaction)
     {
         return TransactionResource::make($transaction);
     }
+
+    public function store(TransactionPostRequest $request)
+    {
+        function processAccounts($payload)
+        {
+
+            $myAccount = Account::findOrFail($payload['account_id']);
+            $myLatestTransaction = getLatestTransaction($myAccount->id);
+
+            if ($payload['transaction_type'] !== 'TRANSFER') {
+
+                return [
+                    [
+                        'account_id' => $myAccount->id,
+                        'user_id' => $myAccount->user_id,
+                        'starting_balance' => getStartingBalance($myLatestTransaction),
+                    ],
+                ];
+            }
+
+            $recAccount = Account::findOrFail($payload['receiver_id']);
+            $recLatestTransaction = getLatestTransaction($recAccount->id);
+
+            return [
+                [
+                    'account_id' => $myAccount->id,
+                    'user_id' => $myAccount->user_id,
+                    'starting_balance' => getStartingBalance($myLatestTransaction),
+                ],
+                [
+                    'account_id' => $recAccount->id,
+                    'user_id' => $recAccount->user_id,
+                    'starting_balance' => getStartingBalance($recLatestTransaction),
+                ],
+            ];
+
+        }
+
+        //START
+
+        $payload = $request->collect();
```

Comment:
collect only retrieves the data as a collection, no validation is happening here
validate input first

Reply:
@IanUrriza10 
use proper safety measures when accessing input data 

https://laravel.com/docs/8.x/validation#working-with-validated-input